utils/s3/client.cc

```diff
@@ -1171,10 +1171,14 @@ future<> client::upload_file(std::filesystem::path path,
                               sstring object_name,
                               upload_progress& up,
                               seastar::abort_source* as) {
+    // The default part size for multipart upload is set to 50MiB.
+    // This value was determined empirically by running `perf_s3_client` with various part sizes to find the optimal one.
+    static constexpr size_t default_part_size = 50_MiB;
+
     do_upload_file do_upload{shared_from_this(),
                              std::move(path),
                              std::move(object_name),
-                             {}, 0, up, as};
+                             {}, default_part_size, up, as};
```

Comment:
I think it's the wrong place, as there's upload_file overload that may still pass 0 to do_upload_file consturctor and it will fallback to 5MiB. The correct place is calc_part_size which should default to 50MiB minimum